src/main/scala/akka/stream/alpakka/xlsx/Row.scala

```diff
@@ -2,31 +2,31 @@ package akka.stream.alpakka.xlsx
 
 import scala.collection.mutable
 
-final class Row(
-    val rowNum: Int,
-    private val cellMap: mutable.TreeMap[Int, Cell]
-) {
-
-  def lastCellNum: Option[Int] = {
-    cellMap.lastOption.map(_._1)
-  }
-
-  def getSize: Int = cellMap.size
-
-  def getCell(cellNum: Int): Option[Cell] = {
-    cellMap.get(cellNum)
-  }
-
-  def simpleCells: Iterable[Cell] = cellMap.values
-
-  lazy val cells: Seq[Cell] = {
-    lastCellNum.map { max =>
-      for (index <- 1 to max) yield {
-        cellMap.getOrElse(index, Cell.Blank(CellReference.generateRef(rowNum, index)))
-      }
-    }.getOrElse(Nil)
-  }
-
-  override def toString: String = s"Row($rowNum, $cells)"
-
+final class Row(val rowIndex: Int, columnIndexToCell: mutable.TreeMap[Int, Cell]) {
+  require(columnIndexToCell.nonEmpty)
+
+  /**
+    * An `Iterable` of all non-`Blank` cells. Please note that values contained by this `Iterable` are guaranteed to be
+    * returned according to ascending order of column numbers, however two consecutive items do not necessarily
+    * represent cells of adjacent columns.
+    */
+  lazy val simpleCells: Iterable[Cell] = columnIndexToCell.values
+
+  /**
+    * A `Sequence` of all cells ordered by ascending column numbers. Two consecutive items of this `Seq` correspond to
+    * cells of adjacent columns. If a cell is missing in the `columnIndexToCell` mapping, a `Blank` cell is created with
+    * reference generated by `CellReference#generateReference`.
+    */
+  lazy val cells: Seq[Cell] = (1 to columnIndexToCell.keys.max).iterator
```

Comment:
These 2 methods may make more sense as `def`s. Especially if the map is (and really has to be) mutable.